packages/widgets/user-management-widget/src/lib/widget/mixins/initMixin/initComponentsMixins/initGenericFlowButtonMixin.ts

```diff
@@ -0,0 +1,173 @@
+import {
+  GenericFlowButtonDriver,
+  FlowDriver,
+  ModalDriver,
+} from '@descope/sdk-component-drivers';
+import {
+  compose,
+  createSingletonMixin,
+  withMemCache,
+} from '@descope/sdk-helpers';
+import { formMixin, loggerMixin, modalMixin } from '@descope/sdk-mixins';
+import { flowSyncThemeMixin } from '../../flowSyncThemeMixin';
+import { initWidgetRootMixin } from './initWidgetRootMixin';
+import { stateManagementMixin } from '../../stateManagementMixin';
+import { createFlowTemplate } from '../../../../helpers';
+import {
+  getEnableOneOrMore,
+  getEnableOnlyOne,
+  getSelectedUsersMap,
+  getSelectedUsersUserIds,
+} from '../../../state/selectors';
+
+export const initGenericFlowButtonMixin = createSingletonMixin(
+  <T extends CustomElementConstructor>(superclass: T) =>
+    class InitGenericFlowButtonMixinClass extends compose(
+      flowSyncThemeMixin,
+      stateManagementMixin,
+      modalMixin,
+      loggerMixin,
+      formMixin,
+      initWidgetRootMixin,
+    )(superclass) {
+      #modal: ModalDriver;
+
+      #flow: FlowDriver;
+
+      #flowButtons: GenericFlowButtonDriver[] = [];
+
+      #modalCallback: (() => void) | null = null;
+
+      #initComponents(ele: Element) {
+        const button = new GenericFlowButtonDriver(ele, {
+          logger: this.logger,
+        });
+        button.disable();
+        button.onClick(() => {
+          this.#initModalContent(button.flowId, button.enableMode);
+        });
+        this.#onIsUserSelectedUpdate(
+          button,
+          getEnableOneOrMore(this.state),
+          getEnableOnlyOne(this.state),
+        );
+        this.#flowButtons.push(button);
+      }
+
+      #initFlowButtons() {
+        const eles = this.shadowRoot?.querySelectorAll(
+          '[data-generic-flow-button-id]',
+        );
+
+        Array.from(eles).forEach((ele) => {
+          this.#initComponents(ele);
+        });
+      }
+
+      #initModal() {
+        this.#modal = this.createModal({ 'data-id': 'generic-flow-modal' });
+        this.#modal.afterClose = () => {
+          if (this.#modalCallback) {
+            const sdk = this.#modal.ele?.querySelector('descope-wc');
+            sdk?.removeEventListener('page-updated', this.#modalCallback);
+            this.#modalCallback = null;
+          }
+        };
+        this.#flow = new FlowDriver(
+          () => this.#modal.ele?.querySelector('descope-wc'),
+          { logger: this.logger },
+        );
+        this.syncFlowTheme(this.#flow);
+      }
+
+      // eslint-disable-next-line class-methods-use-this
+      #onModalNeeded(modal: ModalDriver, sdk: any, callback: () => void) {
```

Comment:
We should either create a class fn and use the class members or create a util fn and pass arguments